### PR TITLE
Feat(pyavd): Add optional dependency pyavd

### PR DESCRIPTION
**Pull request by dlobato**
## Change Summary
Added optional dependency ansible to pyavd that requires ansible avd collection python requirements.

## How to test
Build pyavd package
pip install pyavd-path-to-wheel[ansible] should install all requirements listed at ansible_collections/arista/avd/requirements.txt

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/Makefile [hunk 1]
```diff
@@ -68,6 +68,8 @@ copy-libs: ## Copy files from Ansible AVD collection
 	cp -r $(ANSIBLE_AVD_DIR)/ansible_collections/arista/avd/roles/eos_designs/schemas/eos_designs.schema.pickle $(SCHEMAS_DIR)/
 	cp -r $(ANSIBLE_AVD_DIR)/ansible_collections/arista/avd/plugins/plugin_utils/schema/avd_meta_schema.pickle $(SCHEMAS_DIR)/
 
+	cp $(ANSIBLE_AVD_DIR)/ansible_collections/arista/avd/requirements.txt requirements-ansible-avd.txt
```
Comment (ClausHolbechArista) [inline python-avd/Makefile:71]: You probably need to add a line to fix-libs below to add ansible-core itself. (you can `grep` the line from requirements-dev.txt.

Comment (dlobato) [reply python-avd/Makefile:71]: Done. Make sense to also bring the ansible core dep

File: ansible_collections/arista/avd/docs/pyavd.md [hunk 2]
```diff
@@ -86,6 +86,12 @@ Optional `md-toc` requirement (automatically installed with above command):
 md-toc>=8.1.8
 ```
 
+To install ansible [avd collection additional python requirements](installation/collection-installation.md#additional-python-libraries-required) install with extra `ansible`:
+
+```sh
+pip3 install pyavd[ansible]
+```
+
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/pyavd.md:94]:
```suggestion

This is only a shortcut to install all Python requirements for the `arista.avd` Ansible collection, and will not install PyAVD itself.

```

File: ansible_collections/arista/avd/docs/pyavd.md [hunk 3]
```diff
@@ -86,6 +86,12 @@ Optional `md-toc` requirement (automatically installed with above command):
 md-toc>=8.1.8
 ```
 
+To install ansible [avd collection additional python requirements](installation/collection-installation.md#additional-python-libraries-required) install with extra `ansible`:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/pyavd.md]:
```suggestion
To install Ansible [AVD collection additional Python requirements](installation/collection-installation.md#additional-python-libraries-required) install with extra `ansible`:
```

**Comment by ClausHolbechArista** [comment]
Moving to draft until comments have been evaluated/addressed.

File: ansible_collections/arista/avd/docs/pyavd.md (hunk 2 above)
Comment (dlobato) [reply ansible_collections/arista/avd/docs/pyavd.md:94]: pyavd[ansible] will install PyAVD plus the `arista.avd` collection requirements if I'm not mistaken, I'd remove the last sentence.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/pyavd.md:94]: OK, my bad.

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM tested install locally with: `pip install './python-avd/dist/pyavd-4.8.0.dev0-py3-none-any.whl[ansible]'` and all requirements.txt + ansible-core was included.

```bash
pip install './python-avd/dist/pyavd-4.8.0.dev0-py3-none-any.whl[ansible]' --upgrade
Processing ./python-avd/dist/pyavd-4.8.0.dev0-py3-none-any.whl (from pyavd==4.8.0.dev0)
Requirement already satisfied: jinja2>=3.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (3.1.3)
Requirement already satisfied: jsonschema<4.18,>=4.5.1 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (4.17.3)
Requirement already satisfied: deepmerge>=1.1.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (1.1.1)
Requirement already satisfied: netaddr>=0.7.19 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (1.2.1)
Requirement already satisfied: treelib>=1.5.5 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (1.7.0)
Requirement already satisfied: cvprac>=1.3.1 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (1.3.2)
Requirement already satisfied: requests>=2.27.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (2.31.0)
Requirement already satisfied: PyYAML>=6.0.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (6.0.1)
Requirement already satisfied: md-toc>=7.1.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (8.2.3)
Requirement already satisfied: cryptography>=38.0.4 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (42.0.5)
Requirement already satisfied: aristaproto>=0.1.1 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (0.1.1)
Requirement already satisfied: ansible-core<2.17.0,>=2.14.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (2.16.5)
Requirement already satisfied: packaging in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from ansible-core<2.17.0,>=2.14.0->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (24.0)
Requirement already satisfied: resolvelib<1.1.0,>=0.5.3 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from ansible-core<2.17.0,>=2.14.0->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (1.0.1)
Requirement already satisfied: grpclib<0.5.0,>=0.4.1 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from aristaproto>=0.1.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (0.4.7)
Requirement already satisfied: python-dateutil<3.0,>=2.8 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from aristaproto>=0.1.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (2.9.0.post0)
Requirement already satisfied: typing-extensions<5.0.0,>=4.7.1 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from aristaproto>=0.1.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (4.11.0)
Requirement already satisfied: cffi>=1.12 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from cryptography>=38.0.4->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (1.16.0)
Requirement already satisfied: MarkupSafe>=2.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from jinja2>=3.0->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (2.1.5)
Requirement already satisfied: attrs>=17.4.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from jsonschema<4.18,>=4.5.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (23.2.0)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from jsonschema<4.18,>=4.5.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (0.20.0)
Requirement already satisfied: fpyutils<5,>=4.0.1 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from md-toc>=7.1.0->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (4.0.1)
Requirement already satisfied: charset-normalizer<4,>=2 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from requests>=2.27.0->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from requests>=2.27.0->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (3.6)
Requirement already satisfied: urllib3<3,>=1.21.1 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from requests>=2.27.0->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (2.2.1)
Requirement already satisfied: certifi>=2017.4.17 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from requests>=2.27.0->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (2024.2.2)
Requirement already satisfied: six in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from treelib>=1.5.5->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (1.16.0)
Requirement already satisfied: pycparser in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from cffi>=1.12->cryptography>=38.0.4->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (2.22)
Requirement already satisfied: h2<5,>=3.1.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from grpclib<0.5.0,>=0.4.1->aristaproto>=0.1.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (4.1.0)
Requirement already satisfied: multidict in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from grpclib<0.5.0,>=0.4.1->aristaproto>=0.1.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (6.0.5)
Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from requests[socks]>=2.27.0->cvprac>=1.3.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (1.7.1)
Requirement already satisfied: hyperframe<7,>=6.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from h2<5,>=3.1.0->grpclib<0.5.0,>=0.4.1->aristaproto>=0.1.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (6.0.1)
Requirement already satisfied: hpack<5,>=4.0 in /home/carlbuchmann/.pyenv/versions/3.11.8/lib/python3.11/site-packages (from h2<5,>=3.1.0->grpclib<0.5.0,>=0.4.1->aristaproto>=0.1.1->pyavd==4.8.0.dev0->pyavd==4.8.0.dev0) (4.0.0)
pyavd is already installed with the same version as the provided wheel. Use --force-reinstall to force an installation of the wheel.
```

**Pull request merged**